### PR TITLE
Add optional filter to retrieve AMI list given a build date

### PR DESCRIPTION
**Pull request by lukeseawalker**
Signed-off-by: Luca Carrogu <carrogu@amazon.com>

**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: util/generate-ami-list.py [hunk 1]
```diff
@@ -65,7 +65,11 @@ def get_ami_list_from_ec2(main_region, regions, date, cookbook_git_ref, node_git
         elif cookbook_git_ref and node_git_ref:
             filters.append({"Name": "tag:parallelcluster_cookbook_ref", "Values": ["%s" % cookbook_git_ref]})
             filters.append({"Name": "tag:parallelcluster_node_ref", "Values": ["%s" % node_git_ref]})
-            filters.append({"Name": "name", "Values": ["aws-parallelcluster-*"]})
+            if build_date:
```
Comment (enrico-usai) [inline util/generate-ami-list.py]: minor: you can write:
```python
filters.append({"Name": "name", "Values": ["aws-parallelcluster-*%s" % build_date if build_date else ""]})
```

Comment (lukeseawalker) [reply util/generate-ami-list.py]: Thanks for the suggestion.
Btw the correct line is
```
filters.append({"Name": "name", "Values": ["aws-parallelcluster-*%s" % (build_date if build_date else "")]})
```
otherwise you end up in having a Values="" if build_date is not defined

File: util/generate-ami-list.py [hunk 2]
```diff
@@ -215,6 +218,9 @@ def update_amis_txt(amis_txt_file, amis):
     group2 = parser.add_argument_group("Retrieve instances from EC2 searching by cookbook and node git reference")
     group2.add_argument("--cookbook-git-ref", type=str, help="cookbook git hash reference", required=False)
     group2.add_argument("--node-git-ref", type=str, help="node git hash reference", required=False)
+    group2.add_argument(
```
Comment (enrico-usai) [inline util/generate-ami-list.py]: minor: `type=str` is the default so we can remove it from all the arguments.
minor2: Why did you add `(optional)` only for this parameter?

Comment (lukeseawalker) [reply util/generate-ami-list.py]: because you could not specify it and the latest will be taken
